onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQAttention.cpp

```diff
@@ -339,8 +451,35 @@ class DmlOperatorQAttention : public DmlOperator
         mhaOperatorDesc.MaskFilterValue = kernelCreationContext.GetOptionalAttribute<float>(AttrName::MaskFilterValue, -10'000.0f);
         mhaOperatorDesc.HeadCount = numHeads;
         mhaOperatorDesc.MaskType = maskType;
+        if (hasPast)
+        {
+            presetKeyTensorDesc = TensorDesc::ConstructDefaultTensorDesc(presentTensorDataType, preseKeyOutputShape);
+            namedPresentKeyOutputTensorDesc = presetKeyTensorDesc.GetDmlDesc();
+            presetValueTensorDesc = TensorDesc::ConstructDefaultTensorDesc(presentTensorDataType, preseValueOutputShape);
+            namedPresentValueOutputTensorDesc = presetValueTensorDesc.GetDmlDesc();
+            mhaOperatorDesc.PastKeyTensor = hasPast ? &namedPastKeyOutputTensorDesc : nullptr;
+            mhaOperatorDesc.PastValueTensor = hasPast ? &namedPastValueOutputTensorDesc : nullptr;
+            mhaOperatorDesc.OutputPresentKeyTensor = hasPast ? &namedPresentKeyOutputTensorDesc : nullptr;
+            mhaOperatorDesc.OutputPresentValueTensor = hasPast ? &namedPresentValueOutputTensorDesc : nullptr;
+        }
```

Comment:
Ternary operator is redundant here because these lines are already under `if(hasPast)`.

Reply:
yup, missed to clean up this section